plugin/forward/dnstap.go

```diff
@@ -30,30 +30,31 @@ func toDnstap(f *Forward, host string, state request.Request, opts options, repl
 
 	if t == "tcp" {
 		ta := &net.TCPAddr{IP: net.ParseIP(ip), Port: int(port)}
-		msg.SetQueryAddress(tm, ta)
+		msg.SetQueryAddress(q, ta)
 	} else {
 		ta := &net.UDPAddr{IP: net.ParseIP(ip), Port: int(port)}
-		msg.SetQueryAddress(tm, ta)
+		msg.SetQueryAddress(q, ta)
```

Comment:
`msg.SetQueryAddress()` should be called for response as well. Though, not sure which of `msg.SetResponseAddress()` or `msg.SetQueryAddress()` is better for FQ, FR messages, I can't see such details in https://github.com/dnstap/golang-dnstap/blob/master/dnstap.pb/dnstap.proto comments. Perhaps, we could add both

Reply:
and done. Also unsure, use QueryAddress()